mail_activity_done/models/mail_activity.py

```diff
@@ -92,3 +96,77 @@ def _read_progress_bar(self, domain, group_by, progress_bar):
         """
         domain = expression.AND([domain, [("activity_ids.done", "=", False)]])
         return super()._read_progress_bar(domain, group_by, progress_bar)
+
+    def _search_activity_state(self, operator, value):
+        # Totally override the method.
```

Comment:
why can't you just take the list of ids you get from super and filter them for being done? That adds very little overhead and you don't break modularity

Reply:
...and the inverse for negative operators, your full override doesn't cover that case

Reply:
_search_activity_state returns a list of record ids, if the search is made in res.partner it returns partner_ids, i need to filter the subquery, i don't see a way to make it without making the full query again. 
And we need to filter with done=False allways, even if the search is negative operator, if somebody searchs "activity state is not overdue" or "activity state is today", we need to filter the undone activities in both cases because you want to see records with a particular activity_state based on undone activities.
Am i missing something?

Reply:
Oh, thanks!, i would never tought that solution